activesupport/lib/active_support/cache.rb

```diff
@@ -786,12 +809,71 @@ def save_block_result_to_cache(name, options)
 
     module NullCoder # :nodoc:
       class << self
+        def dump(entry)
+          entry
+        end
+
         def load(payload)
           payload
         end
+      end
+    end
+
+    MARSHAL_PREFIX = "\x04\b".b.freeze # :nodoc:
 
+    module LegacyCoder # :nodoc:
+      class << self
         def dump(entry)
-          entry
+          Marshal.dump(entry)
+        end
+
+        def load(payload)
+          if payload.start_with?(MARSHAL_PREFIX)
+            Marshal.load(payload)
+          else
+            EntryCoder.load(payload)
+          end
+        end
+      end
+    end
+
+    module EntryCoder # :nodoc:
+      UNCOMPRESSED = 0
+      COMPRESSED = 1
+
+      class << self
+        def dump(entry)
+          UNCOMPRESSED.chr << Marshal.dump(entry.pack)
+        end
+
+        def dump_compressed(entry, threshold)
```

Comment:
I'm not super fan of this method, but the reason is that I want users to be able to pass any "standard" `coder`, so anything responding to `load(String) / dump(Object)`.

And since the Cache interface allow to pass `compress: true/false` to `write()`, we need to way to enable or disable compression at all time.